lib/pure/random.nim

```diff
@@ -632,6 +662,17 @@ when isMainModule:
       elif oc > 150:
         doAssert false, "too many occurrences of " & $i
 
+    when false:
```

Comment:
I'd:
* move tests to trandom.nim, and use sane bounds as mentioned in https://github.com/nim-lang/Nim/pull/14725/files#r443086736
* turn `when false` into enabled code, otherwise it doesn't even check that it keeps compiling/working
* if on top of that you want a test meant to be run "by users, not by CI", then you could add a `when defined(nimExampleRandom)`, unless someone has a better idea; there's also $nim/examples but there's value to keeping it colocated where tests are